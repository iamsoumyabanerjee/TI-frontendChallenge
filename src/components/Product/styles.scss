@import '../../styles/breakpoint.scss';
@import '../../styles/variables.scss';

.product-wrapper {
  width: 100%;
  @include viewport(small) {
    width: 50%;
  }
  @include viewport(medium) {
    width: 33.3%;
  }
  @include viewport(large) {
    width: 25%;
  }

  .container {
    // background: linear-gradient(rgba(0, 0, 0, 0.05), rgba(0, 0, 0, 0.05)), #FAF1E4;
    border: 1px solid #CEC7BC;
    border-radius: 16px;
    margin-top: 32px;
    cursor: pointer;
    opacity: 0;
    top: 0;
    animation-duration: 0.3s;
    position: relative;
    margin-left: 16px;
    margin-right: 16px;
    animation-name: fadeIn;
    transition: transform .2s ease-in-out;
    @keyframes fadeIn {
      from { opacity: 0; top: 20px }
      to   { opacity: 1; top: 0 }
    }
    &:hover {
      -webkit-box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
      -moz-box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
      box-shadow: 0px 0px 5px 0px rgba(0,0,0,0.75);
      transform: scale(1.05);
    }
    .product-item {
      color: $nero;
      display: block;
      text-decoration: none;
      position: relative;
      .product-image {
        display: block;
        border-radius: 16px 16px 0 0;
        overflow: hidden;
      }
      .short-details {
        padding: 16px;
        display: flex;
        flex-direction: column;
        .brand {
          color: $silverChalice;
          font-size: 0.875rem;
        }
        .name {
          color: $matterhorn;
        }
      }
      
      .new-arrival-icon {
        position: absolute;
        left: -12px;
        top: -12px;
        height: 30px;
        width: 30px;
      }
    }
    img {
      width: 100%;
    }
  }
}