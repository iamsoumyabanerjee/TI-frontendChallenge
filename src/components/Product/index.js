import React from 'react'

import ShowPrice from '../ShowPrice'
import { ITEM_TYPE_NEW } from '../../utils/constants'
import NewArrivals from '../../icons/newArrivals'
import styles from './styles.scss'

const Product = ({ product, index, prevItemCount }) => {
  const { _embedded: { images, brand }, name, price, final_price, link, ribbon } = product
  const setStyle = (count, prevItemCount) => {
    const style = {}
    if (prevItemCount - 1 > count) {
      return style
    } 
    style.animationDelay = count - (prevItemCount - 1) * 200 + 'ms'
    
    return style
  }

  return (<div className='product-wrapper'>
    <div className='container' style={setStyle(index, prevItemCount)} onAnimationEnd={e=> e.target.style.opacity=1}>
      <a className='product-item' href={`https://www.theiconic.com.au/${link}`}>
        <span className='product-image'><img src={images[0].url} alt={name} /></span>
        <div className='short-details'>
          { ribbon && ribbon.toUpperCase() == ITEM_TYPE_NEW ? <NewArrivals styleClas='new-arrival-icon' /> : null }
          <span className='brand'>{brand.name}</span>
          <span className='name'>{name}</span>
          <ShowPrice price={price} finalPrice={final_price} />
        </div>
      </a>  
    </div>
  </div>)
}

export default Product