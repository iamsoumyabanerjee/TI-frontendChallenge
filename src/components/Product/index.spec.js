import React from 'react'
import renderer from 'react-test-renderer'
import Product from '.'

describe('#Product', () => {
  const props = {
    index: 0,
    product: { 
      _embedded: { 
        images: 'some image url', 
        brand: 'abcd' 
      }, 
      name: 'Name', 
      price: 'Some value', 
      final_price: 'Some value', 
      link: 'link value', 
      ribbon: 'value' 
    }
  }

  it('should render correctly', () => {
    const tree = renderer
      .create(<Product { ...props } />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})