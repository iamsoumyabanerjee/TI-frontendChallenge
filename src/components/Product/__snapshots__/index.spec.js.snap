// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Product should render correctly 1`] = `
<div
  className="product-wrapper"
>
  <div
    className="container"
    onAnimationEnd={[Function]}
    style={
      Object {
        "animationDelay": "NaNms",
      }
    }
  >
    <a
      className="product-item"
      href="https://www.theiconic.com.au/link value"
    >
      <span
        className="product-image"
      >
        <img
          alt="Name"
        />
      </span>
      <div
        className="short-details"
      >
        <span
          className="brand"
        />
        <span
          className="name"
        >
          Name
        </span>
        <div
          className="price-container "
        >
          <span
            className="actual-price"
          >
            $
            Some value
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
